cpp/Platform.Converters/CachingConverterDecorator.h

```diff
@@ -1,15 +1,30 @@
-﻿namespace Platform::Converters
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <unordered_map>
+
+namespace Platform::Converters
 {
-    template <typename ...> class CachingConverterDecorator;
-    template <typename TSource, typename TTarget> class CachingConverterDecorator<TSource, TTarget> : public IConverter<TSource, TTarget>
+    template <typename ...> class Cached;
+    template <typename Function> class Cached<Function>
```

Comment:
```suggestion
    template <typename TFunction, typename TSource, typename TTarget> class Cached<Function>
```
We should add these types back to make it possible to store unordered_map as a class member.